### ISSUE TITLE
Unnecessary assert in AddFunction()?

### ISSUE THREAD

**Issue by CrendKing** [AviSynth/AviSynthPlus#144]
I'm developing an AviSynth DirectShow filter (similar to ffdshow). I found that there is the `AVS_LINKAGE_DLLIMPORT` macro. I'm using it to import `AVS_Linkage` without going through the unnecessary plugin importing logic (because my filter is loading AviSynth, not the other way around). However, in Debug build, when I call `IScriptEnvironment::AddFunction()`, there is an assert in `PluginManager::AddFunction()` that verify if my function is a "script function":
```
if (PluginInLoad != NULL)
{
    newFunc = new AVSFunction(name, PluginInLoad->BaseName.c_str(), params, apply, user_data, PluginInLoad->FilePath.c_str());
}
else
{
    newFunc = new AVSFunction(name, NULL, params, apply, user_data, NULL);
    if(apply != &create_c_video_filter)
        assert(newFunc->IsScriptFunction());
}
```
which has
```
  return ( (apply == &(ScriptFunction::Execute))
		  || (apply == &Eval)
          || (apply == &EvalOop)
          || (apply == &Import)
        );
```
thus fail the assertion. Everything actually works if I comment the assertion or under Release build.

Am I using `AVS_LINKAGE_DLLIMPORT` correctly or this assertion is wrong? If I'm misusing it, what is correct way to get `AVS_Linkage` without loading myself as a plugin in C++ API?

**Comment by pinterf** [comment]
I checked why ffdshow had no problem - I had an abandoned attempt two years ago to add high bit depth to ffdshow, it took way too many weeks to port it even if many filters were omitted), it's because it used the C interface. When a host software using cpp interface is loading avisynth then this breaks the development. So I removed this check. Nevertheless I have never encountered with it.

Resolved in f271e9e34c1deb6cf3dcaf0a67c10079d2f25024
